lms/services/document_url.py

```diff
@@ -32,7 +32,8 @@ def get_document_url(self, request) -> Optional[str]:
 
         return None
 
-    _ENCODED_URL = re.compile("^(?:https?|canvas|vitalsource|jstor)%3a", re.IGNORECASE)
+    # If the URL looks like "schema%3a" (schema:) it's double-encoded
+    _ENCODED_URL = re.compile("^(?:[a-z]+)%3a", re.IGNORECASE)
```

Comment:
We currently only use letters in schemas, but it is conceivable that someone might try to use other schema-allowed characters in future. I'm mainly thinking of digits and "-", but "." and "+" are also technically allowed.